atomix/cluster/src/test/java/io/atomix/raft/ControllableRaftContexts.java

```diff
@@ -310,35 +319,70 @@ public void clientAppendOnLeader() {
     }
   }
 
+  public void snapshotAndCompact(final MemberId memberId) {
+    final RaftContext raftContext = raftServers.get(memberId);
+    // Take snapshot at an index between lastSnapshotIndex and current commitIndex
+    final TestSnapshotStore testSnapshotStore = snapshotStores.get(memberId);
+    final var startIndex =
+        Math.max(raftContext.getLog().getFirstIndex(), testSnapshotStore.getCurrentSnapshotIndex());
+    if (startIndex >= raftContext.getCommitIndex()) {
+      // cannot take snapshot
+      return;
+    }
+    final long snapshotIndex = random.nextLong(startIndex, raftContext.getCommitIndex());
+    try (final RaftLogReader reader = raftContext.getLog().openCommittedReader()) {
+      reader.seek(snapshotIndex);
+      final long term = reader.next().term();
+
+      InMemorySnapshot.newPersistedSnapshot(
+          snapshotIndex, term, random.nextInt(2, 10), testSnapshotStore);
```

Comment:
```suggestion
          snapshotIndex, term, random.nextInt(1, 10), testSnapshotStore);
```

:wrench: Let's also produce snapshots with a single chunk. I think that's an interesting edge case where potential bugs might hide. (Even if we probably never have snapshots with a single chunk in Zeebe)